File: leetcode/leetcode1288.py
#https://leetcode.cn/problems/remove-covered-intervals/?envType=daily-question&envId=2026-07-06
from typing import  List
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x:(x[0],-x[1]))
        n=len(intervals)
        ans=n
        right=intervals[0][1]
        for i in range(1,n):
            if intervals[i][1]<=right:
                ans=ans-1
            else:
                right=intervals[i][1]
        return ans

File: leetcode/leetcode1288.cpp
using namespace std;
#include<vector>
#include<algorithm>
class Solution {
public:
    int removeCoveredIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end(),[](vector<int>&a,vector<int>&b){
            if(a[0]!=b[0]){
                return a[0]<b[0];
            }
            return a[1]>b[1];
        });
        int n=intervals.size();
        int ans=n;
        int right=intervals[0][1];
        for (int i=1;i<n;i++){
            if (intervals[i][1]<=right){
                ans--;
            }else{
                right=intervals[i][1];
            }
        }
        return ans;
    }
};